File: questions/question_1.1/problem_1.1.js
var uniqueChars = function(word){
  var histo = {}
  for (index = 0; index < word.length; index++){
    if (histo[word[index]]){
      histo[word[index]]++
    } else {
      histo[word[index]] = 1
    }
  }

  for (property in histo){
    if(histo[property] > 1){
      return false;
    }
    return true;
  }
};

console.log(uniqueChars('hello'));
console.log(uniqueChars('helro'));

File: questions/question_1.1/problem_1.1.rb
def unique_chars? word
  histogram = word.split('').inject(Hash.new(0)){ |h,v| h[v] += 1; h }
  histogram.each {|k,v| return false if v > 1}
  true
end

p unique_chars? 'hello'

p unique_chars? 'bite'